deps/rabbit/src/rabbit_ff_controller.erl

```diff
@@ -1040,7 +1043,28 @@ merge_feature_flags(FeatureFlagsA, FeatureFlagsB) ->
 
               FeatureProps1 = FeatureProps#{stability => Stability},
               FeatureProps2 = maps:remove(callbacks, FeatureProps1),
-              FeatureProps2
+              FeatureProps2;
+          (FeatureName, FeatureProps) when ?IS_DEPRECATION(FeatureProps) ->
+              UnknownProps = #{deprecation_phase => optout},
+              FeaturePropsA = maps:get(
+                                FeatureName, FeatureFlagsA, UnknownProps),
+              FeaturePropsB = maps:get(
+                                FeatureName, FeatureFlagsB, UnknownProps),
+
+              PhaseA = rabbit_deprecated_features:get_phase(FeaturePropsA),
+              PhaseB = rabbit_deprecated_features:get_phase(FeaturePropsB),
+              Phase = case {PhaseA, PhaseB} of
+                          {removed, _}      -> removed;
+                          {_, removed}      -> removed;
+                          {disconnected, _} -> disconnected;
+                          {_, disconnected} -> disconnected;
+                          {optin, _}        -> optin;
```

Comment:
Should this also take into account the actual config values for opting in/out? Because `optin` and `optout` carry no information about whether the feature is actually disabled on a particular node. 

I.e. hierarchy like `removed` -> `disconnected` -> `opted_out_by_default`(i.e. it's `optin` phase without or with `false` config value) ->  `opted_out_by_choice` (`optout` phase, `true` config value) -> `feature_still_available`.

Reply:
The state of a feature flag or deprecated feature is maintained separately. Here, we are consolidating the properties of a feature flag. In case the same feature flag or deprecated feature has different definitions between nodes, we use this code to prioritize a phase : `removed` > `disconnected` > `optin` > `optout`, regardless of the configuration which affects the actual state, not the definition of the deprecated feature.

Does it make sense?

Reply:
Yes, I didn't notice that actual state is managed separately. I think now I finally understand what's your intention behind it.